recipe/gen_patch_json.py

```diff
@@ -2547,6 +2547,12 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
             new_constrains = record.get("constrains", [])
             new_constrains.append("jpeg <0.0.0a")
             record["constrains"] = new_constrains
+            
+            
+        # Different patch versions of imath can be ABI incompatible
+        # See https://github.com/conda-forge/imath-feedstock/issues/7
+        if has_dep(record, "imath") and record.get('timestamp', 0) < 1678196668497:
+            _pin_stricter(fn, record, "imath", "x.x.x")
```

Comment:
I think we actually want to go back in time, and add a stricter pin on all the packages we released:

```
                    _pin_stricter(fn, record, "imath", "x", upper_bound="3.1.6")
```
i might have a an off-by-one error in the command above.

Reply:
Sorry, i always get repodata patches wrong. I think we want to make sure that things compiled with imath 3.1.0 through 3.1.6 stay usable together, otherwise we would break existing, working environments.

Reply:
Good point, we started packaging from imath 3.1.4, and that one has the same soversion of 3.1.5 and 3.1.6, i.e. 29:
* https://github.com/AcademySoftwareFoundation/Imath/blob/v3.1.4/CMakeLists.txt#L41
* https://github.com/AcademySoftwareFoundation/Imath/blob/v3.1.6/CMakeLists.txt#L41

So just having an upper_bound to 3.1.6 makes more sense.

Reply:
Done in https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/413/commits/719451eae15162addde0a66e46155db164149962, by looking at the diff in https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/413#issuecomment-1458314871 it seems to be working as intended.